plugin/net/watcher.go

```diff
@@ -58,3 +58,31 @@ func (w *watcher) ContainerDied(id string) {
 }
 
 func (w *watcher) ContainerDestroyed(id string) {}
+
+func (w *watcher) ContainerConnected(id string) {
+	w.driver.debug("ContainerConnected", "%s", id)
+	info, err := w.client.InspectContainer(id)
```

Comment:
This code should replace the `ContainerStarted` version, not add to it.

Also it should use the [information inside the event](https://github.com/docker/docker/blob/d8406fd7a029f38ece55421294568045d1c9de92/daemon/events.go#L91) about what network was connected, which avoids having to loop round hunting for it.